### PR TITLE
Convert MultipartBody to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/MultipartBody.kt [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    }
+
+    /** Add a part to the body.  */
+    fun addPart(part: Part) = apply {
+      parts.add(part)
+    }
+
+    /** Assemble the specified parts into a request body.  */
+    fun build(): MultipartBody {
+      require(!parts.isEmpty()) { "Multipart body must have at least one part." }
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/MultipartBody.kt]:
```suggestion
      require(parts.isNotEmpty()) { "Multipart body must have at least one part." }
```

File: okhttp/src/main/java/okhttp3/MultipartBody.kt [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+     * Appends a quoted-string to a StringBuilder.
+     *
+     * RFC 2388 is rather vague about how one should escape special characters in form-data
+     * parameters, and as it turns out Firefox and Chrome actually do rather different things, and
+     * both say in their comments that they're not really sure what the right approach is. We go
+     * with Chrome's behavior (which also experimentally seems to match what IE does), but if you
+     * actually want to have a good chance of things working, please avoid double-quotes, newlines,
+     * percent signs, and the like in your field names.
+     */
+    internal fun appendQuotedString(target: StringBuilder, key: String) {
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/MultipartBody.kt]: Could be extension if you want

File: okhttp/src/main/java/okhttp3/MultipartBody.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/MultipartBody.kt]: Fixed.

File: okhttp/src/main/java/okhttp3/MultipartBody.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/MultipartBody.kt]: Good call. Fixed.

**Comment by JakeWharton** [comment]
Uh. Something weird happened here. Rebase?

**Comment by JakeWharton** [comment]
Oh you just sorted the list. If you want to be cute we can add a check to make sure it's sorted.

File: okhttp/src/main/java/okhttp3/MultipartBody.kt [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        return Part(headers, body)
+      }
+
+      @JvmStatic
+      fun createFormData(name: String, value: String): Part =
+          createFormData(name, null, RequestBody.create(null, value))
+
+      @JvmStatic
+      fun createFormData(name: String, filename: String?, body: RequestBody): Part {
+        val disposition = StringBuilder("form-data; name=")
```
Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/MultipartBody.kt]: Could be `buildString`, or we can clean them all up later

Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/MultipartBody.kt]: Later!

**Pull request merged**